### PR TITLE
Otto-App UI Enhancements Pt2.

### PR DESCRIPTION
**Pull request by darkbrewery**

### CODE REVIEW

**Comment by darkbrewery** [comment]
Looks like ImageMin broke the yarn build, commenting it out until devs check dependencies are installed, am unable to test from my end. @pyc92 @0xthk

**Comment by darkbrewery** [comment]
With these commits, all TypeScript errors will be resolved

**Comment by darkbrewery** [comment]
@pyc92 This is all ready to go now!

**Comment by darkbrewery** [comment]
@pyc92 @Appppo added new layout, page and view for palace, to embed notion in an iframe. This prevents traffic from leaving the app and maintains the user experience.

**Comment by darkbrewery** [comment]
Cleaned up Iframe code so we have only one view and one layout for two new pages

**Comment by darkbrewery** [comment]
@pyc92 this is all ready to go again!

File: .gitignore [hunk 1]
```diff
@@ -32,3 +32,8 @@ tsconfig.tsbuildinfo
 
 # generated files
 public/locales/zh-cn/*.json
+
+# local yarn install
+yarn.lock
```
Comment (pyc92) [inline .gitignore:37]: yarn lock should not be ignored

File: src/views/otto/OttoPage/index.tsx [hunk 2]
```diff
@@ -360,16 +360,19 @@ export default function OttoPage() {
...
+      STR: '/trait-icons/STR.png',
+      DEF: '/trait-icons/DEF.png',
+      DEX: '/trait-icons/DEX.png',
+      INT: '/trait-icons/INT.png',
+      LUK: '/trait-icons/LUK.png',
+      CON: '/trait-icons/_CON.png',
+      CUTE: '/trait-icons/CUTE.png',
+    }),
+    []
+  )
```
Comment (pyc92) [inline src/views/otto/OttoPage/index.tsx:375]: `AttrIconsMap` is a const var, it's better to move it to the global scope.

**Comment by pyc92** [comment]
there are lots of typing errors, here is the details:
https://github.com/OtterClam/otto-app/actions/runs/4361336687/jobs/7625141669

**Comment by darkbrewery** [comment]
@pyc92 these should all be resolved now, typing errors were fixed in this commit:
https://github.com/OtterClam/otto-app/pull/400/commits/d911c5a9707039eccd39e4326191df5580e209f7

**Comment by pyc92** [comment]
@darkbrewery I think that's just one of the errors. you can run `yarn check-types` locally.

Screen Shot 2023-03-12 at 3 07 01 PM

Screen Shot 2023-03-12 at 3 07 16 PM

**Comment by darkbrewery** [comment]
Yes, they've all been resolved

image

**Comment by darkbrewery** [comment]
https://github.com/OtterClam/otto-app/pull/400/commits/d911c5a9707039eccd39e4326191df5580e209f7

**Comment by pyc92** [comment]
cool!
The last thing to note is that the yarn.lock file should not be ignored by Git because we need this file to ensure that the versions of dependencies installed on the CI are the same as those installed on our local machine.

**Comment by darkbrewery** [comment]
@pyc92 Alrighty, yeah makes sense, I've taken it out of the gitignore here https://github.com/OtterClam/otto-app/pull/400/commits/d86944bb7f39aa04a7d7c4882aa14634f1f9d29b

**Comment by darkbrewery** [comment]
@pyc92 after this merge, I'll start on another pull to clean up all the lint warnings, but it's all running great on my end

**Pull request merged**